### PR TITLE
Add logic that checks for number of crashes on disk. Will skip crash …

### PR DESCRIPTION
**Pull request by TroubleMakerBen**
…if more than 50 chrashes

### CODE REVIEW

File: hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java [hunk 1]
```diff
@@ -75,6 +74,15 @@ public static void saveException(Throwable exception, Thread thread, CrashManage
             return;
         }
 
+        // Check for number of crashes on disk and don't save the crash in case we have more than 25 on disk.
```
Comment (dhei) [inline hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java]: The limit is 50 here. Should update this comment and Line81.

File: hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java [hunk 2]
```diff
@@ -75,6 +74,15 @@ public static void saveException(Throwable exception, Thread thread, CrashManage
...
+        final String[] list = CrashManager.searchForStackTraces(CrashManager.weakContext);
+        HockeyLog.debug("ExceptionHandler: Found " + list.length + " stacktrace(s).");
+        if(list.length >= 50) {
+            HockeyLog.warn("ExceptionHandler: HockeyApp will not save this exception as there are already 25 unsent " +
```
Comment (dhei) [inline hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java]: Can we merge Line81 and Line82 into one string?

File: hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java (hunk 1 above)
Comment (guperrot) [reply hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java]: A constant could avoid further mistakes in future

File: hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java [hunk 3]
```diff
@@ -75,12 +75,12 @@ public static void saveException(Throwable exception, Thread thread, CrashManage
             return;
         }
 
-        // Check for number of crashes on disk and don't save the crash in case we have more than 50 on disk.
+        // Check for number of crashes on disk and don't save the crash in case we have 50 or more files on disk.
```
Comment (guperrot) [inline hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java]: Replace
> 50 or more files
to
> reached the defined limit

File: hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java [hunk 4]
```diff
@@ -75,12 +75,12 @@ public static void saveException(Throwable exception, Thread thread, CrashManage
...
         final String[] list = CrashManager.searchForStackTraces(CrashManager.weakContext);
         final int number = list.length;
         HockeyLog.debug("ExceptionHandler: Found " + number + " stacktrace(s).");
         if(number >= MAX_NUMBER_OF_CRASHFILES) {
-            HockeyLog.warn("ExceptionHandler: HockeyApp will not save this exception as there are already 50 unsent exceptions on disk");
+            HockeyLog.warn("ExceptionHandler: HockeyApp will not save this exception as there are already 50 or more unsent exceptions on disk");
```
Comment (guperrot) [inline hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java]: use constant for 50

File: hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java [hunk 5]
```diff
@@ -75,6 +75,15 @@ public static void saveException(Throwable exception, Thread thread, CrashManage
             return;
         }
 
+        // Check for number of crashes on disk and don't save the crash in case we reached the defined limit.
+        final String[] list = CrashManager.searchForStackTraces(CrashManager.weakContext);
+        final int number = list.length;
+        HockeyLog.debug("ExceptionHandler: Found " + number + " stacktrace(s).");
+        if(number >= MAX_NUMBER_OF_CRASHFILES) {
+            HockeyLog.warn("ExceptionHandler: HockeyApp will not save this exception as there are already" + MAX_NUMBER_OF_CRASHFILES + "or more unsent exceptions on disk");
```
Comment (jwallra) [inline hockeysdk/src/main/java/net/hockeyapp/android/ExceptionHandler.java]: (nit) Pad MAX_NUMBMER_OF_CRASHFILES with spaces

File: hockeysdk/src/main/java/net/hockeyapp/android/CrashManager.java [hunk 6]
```diff
@@ -793,7 +793,7 @@ private static void saveConfirmedStackTraces(final WeakReference<Context> weakCo
     /**
      * Searches .stacktrace files and returns them as array.
      */
-    private static String[] searchForStackTraces(final WeakReference<Context> weakContext) {
+    protected static String[] searchForStackTraces(final WeakReference<Context> weakContext) {
```
Comment (MatkovIvan) [inline hockeysdk/src/main/java/net/hockeyapp/android/CrashManager.java]: I guess we should apply limit here (on sending) as well.
For correct processing existing crashes, I suggest take only 50 oldest crashes and delete others without sending.

Comment (TroubleMakerBen) [reply hockeysdk/src/main/java/net/hockeyapp/android/CrashManager.java]: I suggest doing this as a separate PR to keep the PRs small.

**Pull request merged**